server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -291,21 +291,26 @@ public class Node implements Closeable {
     final NamedWriteableRegistry namedWriteableRegistry;
     final NamedXContentRegistry namedXContentRegistry;
 
+    /**
+     * Constructs a node
+     *
+     * @param environment         the initial environment for this node, which will be added to by plugins
+     */
     public Node(Environment environment) {
-        this(environment, Collections.emptyList(), true);
+        this(environment, PluginsService.getCreatePluginsServiceFunction(environment), true);
     }
 
     /**
      * Constructs a node
      *
      * @param initialEnvironment         the initial environment for this node, which will be added to by plugins
-     * @param classpathPlugins           the plugins to be loaded from the classpath
+     * @param pluginsServiceFunction     a function that takes a {@link Settings} object and returns a {@link PluginsService}
```

Comment:
Instead of the generic "function" let's call this the ctor? ie `pluginServiceCtor`